File: app/src/main/java/com/example/fang/b16traveldomain/seatsavailable/SeatsAvailableContract.java
package com.example.fang.b16traveldomain.seatsavailable;

import com.example.fang.b16traveldomain.model.dataresource.busInformation.BusInformation;
import com.example.fang.b16traveldomain.model.dataresource.seatinformation.Seats;

public interface SeatsAvailableContract {

    interface SeatsAvailableView{
        void setupSeatAvailableMVP();
        void showSeatAvailableActivity(Seats seats, BusInformation busInformation);
        void setupSeatsView();
        void getSeats();
    }

    interface SeatsAvailablePresenter{
        void findSeats(BusInformation mbus);
    }
}

File: app/src/main/java/com/example/fang/b16traveldomain/seatsavailable/SeatsAvailableContract.kt
package com.example.fang.b16traveldomain.seatsavailable

import com.example.fang.b16traveldomain.model.dataresource.busInformation.BusInformation
import com.example.fang.b16traveldomain.model.dataresource.seatinformation.Seats

interface SeatsAvailableContract {

    interface SeatsAvailableView {
        fun setupSeatAvailableMVP()
        fun showSeatAvailableActivity(seats: Seats, busInformation: BusInformation)
        fun setupSeatsView()
        fun getSeats()
    }

    interface SeatsAvailablePresenter {
        fun findSeats(mbus: BusInformation)
    }
}
